deployment/ansible/deploy.yml

```diff
@@ -0,0 +1,70 @@
+---
+- name: Deploy epoch package
+  hosts: all
+  remote_user: epoch
+
+  vars:
+    project_root: "{{ ansible_env.HOME }}/node"
+
+  tasks:
+    - name: "Required variables"
+      fail: msg="Either remote_package or local_package variables must be defined"
+      when: remote_package is not defined and local_package is not defined
+
+    - name: "Make sure {{ project_root }} exists"
+      file:
+          path: "{{ project_root }}"
+          state: directory
+          mode: 0700
+
+    - name: Download epoch package
+      get_url:
+        url: "{{ remote_package }}"
+        dest: "{{ project_root }}/{{ remote_package | basename }}"
+      when: remote_package is defined
+
+    - name: Check if epoch binary exists
```

Comment:
I am sure I am missing something - how can this step work before "Extract ..."?

Reply:
Extract steps are at lines 43 and 51.

Reply:
This actually checks if the package was deployed already and there is bin/epoch, then it execute ping and if "pong" it stops the daemon.

If the binary does not exists, it means "first deploy" and no package and daemon yet, so we don't need the ping and stop commands.